src/DynamoPackages/PackageLoader.cs

```diff
@@ -397,29 +374,61 @@ public void LoadPackages(IEnumerable<Package> packages)
                 }
             }
 
-            var assemblies = packageList
+            var assemblies = packages
                 .SelectMany(p => p.LoadedAssemblies.Where(y => y.IsNodeLibrary))
                 .Select(a => a.Assembly)
                 .ToList();
+
             OnPackagesLoaded(assemblies);
         }
 
-        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        private void LoadNewPackages(LoadPackageParams loadPackageParams)
         {
-            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            foreach (var path in loadPackageParams.NewPaths)
             {
-                customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
-                if (!packagesDirectories.Contains(path))
+                var packagesDirectory = pathManager.PackagesDirectories.Where(x => x.StartsWith(path)).FirstOrDefault();
+                ScanPackageDirectories(packagesDirectory, loadPackageParams.Preferences);
+            }
+
+            if (pathManager != null)
+            {
+                foreach (var pkg in LocalPackages)
                 {
-                    packagesDirectories.Add(path);
+                    if (Directory.Exists(pkg.BinaryDirectory))
+                    {
+                        pathManager.AddResolutionPath(pkg.BinaryDirectory);
+                    }
+
                 }
             }
-            LoadAll(loadPackageParams);
+
+            if (LocalPackages.Any())
+            {
+                LoadPackages(LocalPackages);
+            }
+        }
+
+        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        {
+            var packages = new List<Package>(localPackages);
+            foreach(var package in packages)
+            {
+                localPackages.Remove(package);
```

Comment:
what is this doing ?
Your creating a copy of the localPackages.
Then you are removing all the copies from the original localPackages ?
What about a localPackages.clear() or something similar ?

Reply:
Yes, I think I can use clear(). I need to make a copy in `packages` as I need to remember this list so as to add it back in line 426.

Reply:
yeah seems to deserve a comment

Reply:
because the diff is hard to read this is the previous code:
```
 public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
        {
            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
            {
                customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
                if (!packagesDirectories.Contains(path))
                {
                    packagesDirectories.Add(path);
                }
            }
            LoadAll(loadPackageParams);
        }
```

Reply:
but - whats the point? To avoid duplicates?

Reply:
Yes, I'm first removing packages from `localPackages` to avoid adding duplicates. Then I'm adding them back to keep this list current as at all times it need to maintain the currently loaded list of packages.